### PR TITLE
3 dimensional input for packed bf16

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**
Even round-trippability seems broken with this, I see no errors or outputs on running standalone-opt on the committed mlir file.

### CODE REVIEW

File: lib/Standalone/ConvertTppToXsmm.cpp [hunk 1]
```diff
@@ -47,11 +47,17 @@ struct ConvertTppMatmulOp : public OpRewritePattern<MatmulOp> {
     MemRefType memrefB = matmulOp.getMatrixBType();
     int64_t m = memrefC.getShape()[0];
     int64_t n = memrefC.getShape()[1];
-    int64_t k = memrefA.getShape()[1];
+    int64_t k = memrefB.getShape()[0];
```
Comment (chelini) [inline lib/Standalone/ConvertTppToXsmm.cpp:50]: why this change?

File: lib/Standalone/ConvertTppToXsmm.cpp [hunk 2]
```diff
@@ -47,11 +47,17 @@ struct ConvertTppMatmulOp : public OpRewritePattern<MatmulOp> {
...
     auto ldaDim = getLeadingDim(memrefA);
     if (failed(ldaDim))
       return failure();
     int64_t lda = *ldaDim;
+    if (memrefA.getShape().size() == 3) {
```
Comment (chelini) [inline lib/Standalone/ConvertTppToXsmm.cpp:55]: Should we also check the element type in the memref?

File: lib/Standalone/Dialect/Tpp/TppOps.cpp [hunk 3]
```diff
@@ -56,14 +56,15 @@ LogicalResult MatmulOp::verify() {
   MemRefType a = getMatrixA().getType().cast<MemRefType>();
   MemRefType b = getMatrixB().getType().cast<MemRefType>();
   MemRefType c = getMatrixC().getType().cast<MemRefType>();
-  if ((a.getShape().size() != 2) || (b.getShape().size() != 2) ||
-      (c.getShape().size() != 2))
-    return failure();
+  if ((a.getShape().size() != 2 && a.getShape().size() != 3) ||
```
Comment (chelini) [inline lib/Standalone/Dialect/Tpp/TppOps.cpp:59]: Would you mind also adding constraints on the element type?

File: lib/Standalone/ConvertTppToLoops.cpp [hunk 4]
```diff
@@ -220,6 +220,8 @@ struct ConvertTppMatmulOp : public OpRewritePattern<MatmulOp> {
     ArrayRef<int64_t> shapeC = matmulOp.getMatrixCType().getShape();
     ArrayRef<int64_t> shapeA =
         matmulOp.getMatrixA().getType().cast<MemRefType>().getShape();
+    if (shapeA.size() == 3)
+      matmulOp.emitError("Packed BF16 loops unsupported");
```
Comment (chelini) [inline lib/Standalone/ConvertTppToLoops.cpp:224]: Can you please add a test to check also this exit? thx/

**Comment by rengolin** [comment]
For some reason, on my x86_64 machine, the two bf16 tests are `UNSUPPORTED`. I had to remove the unsupported line from the tests to run (and they pass).

File: lib/Standalone/ConvertTppToXsmm.cpp (hunk 1 above)
Comment (KavithaTipturMadhu) [reply lib/Standalone/ConvertTppToXsmm.cpp:50]: This is a stray change, I will revert this.
